doc/lsst.pipe.base/command-line-task-data-repo-howto.rst

```diff
@@ -0,0 +1,136 @@
+.. _command-line-task-data-repo-howto:
+
+#################################################################
+Using Butler data repositories and reruns with command-line tasks
+#################################################################
+
+:ref:`Command-line tasks <using-command-line-tasks>` use Butler data repositories for reading and writing data.
+Butler data repositories are specially-configured directories on a filesystem.
```

Comment:
>  Will the environment variable facility still work with non-POSIX repositories? Instead of a filesystem path the environment variables can be a general URI?

Certain specific URI forms are/will be supported, not any URI in general.

>  Do we have a facility for deleting re-runs in non-POSIX repositories? The advice I've gotten is that folks `ls` inside the rerun directory and delete un-needed experimental reruns.

We currently have no utilities for manipulating repositories. Subsetting, relocating, removing etc. all currently need to be done external to the Butler.  Some of these are conceptually simple but just haven't been done; others have been waiting on a comprehensive view of repository management (intended to be supported in part by a global metadata service).

Creating a repository should generally be done by executing a `CmdLineTask`.  The problem is two-fold: `CmdLineTask`s currently infer their output repository configuration from their input, so there's a question of how to create the first input; and `CmdLineTask` has no argument to allow specification of Butler configuration information.  I think this lack of progress on integrating `CmdLineTask` and Butler has come about because of reluctance to invest in `CmdLineTask` when it is soon to be replaced by Activators.